### PR TITLE
chore: bump version, update ABI after v2.4.0 release

### PR DESCRIPTION
**Pull request by devbww**
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/10161)

### CODE REVIEW

**Review by coryan — APPROVED** [review]
Thanks for adding the new GA library baselines, it is easy to forget.

File: CMakeLists.txt [hunk 1]
```diff
@@ -21,7 +21,7 @@ set(PACKAGE_BUGREPORT "https://github.com/googleapis/google-cloud-cpp/issues")
 
 project(
     google-cloud-cpp
-    VERSION 2.4.0
+    VERSION 2.5.0
```
Comment (dbolduc) [inline CMakeLists.txt:24]: Did we want to try `2.5.0-rc` ?

Comment (dbolduc) [reply CMakeLists.txt:24]: Also do you want to add the new version to the CHANGELOG as a placeholder in this PR?

Comment (devbww) [reply CMakeLists.txt:24]: > Did we want to try `2.5.0-rc` ?

I'll be working on that separately.  Yes, that means we won't be able to trust "-rc" until 2.6.

Comment (devbww) [reply CMakeLists.txt:24]: > Also do you want to add the new version to the CHANGELOG as a placeholder in this PR?

Oops.  Hoist by my own petard.  Fixed.

**Pull request merged**